validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorConfig.java

```diff
@@ -259,6 +267,11 @@ public Builder graffiti(Bytes32 graffiti) {
       return this;
     }
 
+    public Builder graffitiFile(Path graffitiFile) {
+      this.graffitiFile = graffitiFile;
+      return this;
+    }
```

Comment:
I think we could simplify and make this more easily customizable by defining a `GraffitiProvider` interface:
```
@FunctionalInterface
interface GraffitiProvider {
  Optional<Bytes32> get();
}
```

`ValidatorConfig` could then just have a single `graffitiProvider(...)` setter.

We could then move the new `Graffiti` class to `FileBackedGraffitiProvider implements GraffitiProvider`.   The actual provider instance would be constructed in `ValidatorOptions`.

Reply:
I'm unsure 😕 I think that would yield some different behaviour if the graffiti were to be removed at some point (the fallback behaviour changes). I guess we need to pin that down first?

Reply:
Not suggesting any different functionality here, just an interface to simplify the dependencies.  So we can pass around 1 value (`graffitiProvider`) instead of 2 (`graffiti` and `graffitiFile`).

Reply:
I'm not sure I'm totally following at this point. I'll make some changes and see if that's what you mean.

Reply:
Happy to make a quick patch to show you what I mean as well - let me know if you think that would be helpful